flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/reconciler/deployment/AbstractFlinkResourceReconciler.java

```diff
@@ -180,13 +190,28 @@ public void reconcile(FlinkResourceContext<CR> ctx) throws Exception {
         }
     }
 
-    private void applyAutoscaler(FlinkResourceContext<CR> ctx) throws Exception {
+    private void applyAutoscaler(FlinkResourceContext<CR> ctx, @Nullable String existingOverrides)
+            throws Exception {
         var autoScalerCtx = ctx.getJobAutoScalerContext();
         boolean autoscalerEnabled =
                 ctx.getResource().getSpec().getJob() != null
                         && ctx.getObserveConfig().getBoolean(AUTOSCALER_ENABLED);
         autoScalerCtx.getConfiguration().set(AUTOSCALER_ENABLED, autoscalerEnabled);
+
         autoscaler.scale(autoScalerCtx);
+
+        // Check that the overrides actually changed and not merely the String representation
+        var flinkConfig = ctx.getResource().getSpec().getFlinkConfiguration();
+        var newOverrides = flinkConfig.get(PipelineOptions.PARALLELISM_OVERRIDES.key());
+        if (existingOverrides != null && newOverrides != null) {
```

Comment:
```suggestion
        if (existingOverrides != null && newOverrides != null && !existingOverrides.equals(newOverrides)) {
```

When `existingOverrides.equals(newOverrides)`, we don't need to execute this logic. So we can add one condition here.

Reply:
The rewritten version using Gyula's suggestions doesn't require this anymore.